pkg/controller/template/template_controller.go

```diff
@@ -103,6 +112,65 @@ func New(
 	return ctrl
 }
 
+func (ctrl *Controller) addSecret(obj interface{}) {
+	s := obj.(*v1.Secret)
+	if s.DeletionTimestamp != nil {
+		ctrl.deleteSecret(s)
+		return
+	}
+	if s.GetNamespace() == "openshift-config" && s.GetName() == "pull-secret" {
+		cfg, err := ctrl.ccLister.Get(common.ControllerConfigName)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get ControllerConfig on secret callback %#v", err))
+			return
+		}
+		glog.V(4).Infof("Re-syncing ControllerConfig %s due to secret change", cfg.Name)
+		ctrl.enqueueControllerConfig(cfg)
+	}
+}
+
+func (ctrl *Controller) updateSecret(old, new interface{}) {
+	s := new.(*v1.Secret)
+
+	if s.GetNamespace() == "openshift-config" && s.GetName() == "pull-secret" {
+		cfg, err := ctrl.ccLister.Get(common.ControllerConfigName)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get ControllerConfig on secret callback %#v", err))
+			return
+		}
+		glog.V(4).Infof("Re-syncing ControllerConfig %s due to secret change", cfg.Name)
```

Comment:
do we want all of these logs in the v4 logs? Since this doesn't happen often I'm fine with it being in the main logs

Reply:
I'm fine changing this although I'll be more confident doing it in another PR and probably after 4.1 to avoid some weird logs flooding before we get there (we can discuss this in an issue maybe if we really want it for 4.1?)

Reply:
sure. i just figured since this is something user initiated, we'd want to expose some sort of log. all of the MC based updates for ex result in a log line exposed